scripts/windows/UninstallWindowsAgent.ps1

```diff
@@ -91,25 +177,56 @@ function Remove-JumpCloud {
                 }
             }
         }
+    }
 
-        # Stopping and removing the JumpCloud Agent service
-        $serviceName = "jumpcloud-agent"
-        if (Get-Service -Name $serviceName -ErrorAction SilentlyContinue) {
-            $foundItemsToRemove = $true
-            Stop-Service -Name $serviceName -Force -ErrorAction SilentlyContinue
-            Start-Sleep -Seconds 6
-            sc.exe delete $serviceName
-            Start-Sleep -Seconds 6
-            Write-Host "Service $serviceName successfully removed."
-        } 
-
-        # Removing the JumpCloudfolder
-        $jumpcloudFolder = "C:\Program Files\JumpCloud"
-        if (Test-Path $jumpcloudFolder) {
-            $foundItemsToRemove = $true
-            Remove-Item -Path $jumpcloudFolder -Recurse -Force -ErrorAction SilentlyContinue
-            Write-Host "Successfully deleted folder: $jumpcloudFolder"
-        } 
+    # Stopping and removing the JumpCloud Agent service if exists
+    $serviceName = "jumpcloud-agent"
+    if (Get-Service -Name $serviceName -ErrorAction SilentlyContinue) {
+        $foundItemsToRemove = $true
+        Stop-Service -Name $serviceName -Force -ErrorAction SilentlyContinue
+        Start-Sleep -Seconds 6
+        sc.exe delete $serviceName
+        Start-Sleep -Seconds 6
+        Write-Host "Service $serviceName successfully removed."
+    }
+
+    # stop and remove jumpcloud try if it exists
+    Get-Process -Name jumpcloudtray -ErrorAction SilentlyContinue | ForEach-Object {
+        Write-Host "stopping tray process $($_.Id)"
+        Stop-Process -id $_.Id -ErrorAction SilentlyContinue
+    }
+
+    # Removing the JumpCloudfolder
+    $folder = "$($env:ProgramFiles)\JumpCloud"
+    Remove-Folder $folder
+
+    # remove tray folder
+    $folder = "$($env:ProgramFiles)\JumpCloudTray"
+    Remove-Folder $folder
+
+    $folder = "$env:APPDATA\JumpCloud"
+    Remove-Folder $folder
+
+    # remove older credential provider files if they exists in sys dir
+    if (Test-Path "$env:SystemRoot\System32\JumpCloud*.dll") {
+        $foundItemsToRemove = $true
+        Remove-Item -Path "$env:SystemRoot\System32\JumpCloud*.dll" -Recurse -Force -ErrorAction SilentlyContinue
+        Write-Host "Successfully deleted files from system32"
+    }
```

Comment:
I feel that maybe we want to handle the error conditions in this block and not use `-ErrorAction SilentlyContinue`. It is better to know we failed in the script than to say that the script was successful to later fail in the installation of the agent.